### ISSUE TITLE
Detect illegal use of type annotation on `global` or `nonlocal` symbol

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#10709]
The use of a type annotation on a variable with `global` or `nonlocal` binding results in a "syntax error" at runtime, but pyright does not detect or report this error.

```python
def func():
    global x
    x: str  # Runtime error
```
